### ISSUE TITLE
Invalid index error

### ISSUE THREAD

**Issue by kumar-dhanagopal** [oracle-terraform-modules/terraform-oci-oke#89]
I cloned this repo this morn, set the variables in tfvars, and got a few `Invalid index` errors when I ran `terraform plan`. 

My TF version is 0.12.13.

I hadn't seen these errors last week with the same repo and TF version! What might the issue be?

```
Error: Invalid index

  on modules\oke\nodepools.tf line 26, in resource "oci_containerengine_node_pool" "nodepools":
  26:   node_image_id = var.node_pools.node_pool_image_id == "NONE" ? data.oci_core_images.latest_images[count.index].images[0].id : var.node_pools.node_pool_image_id
    |----------------
    | count.index is 0
    | data.oci_core_images.latest_images is tuple with 1 element

The given key does not identify an element in this collection value.


Error: Invalid index

  on modules\policies\locals.tf line 10, in locals:
  10:   policy_statement = "Allow dynamic-group ${oci_identity_dynamic_group.oke-kms-cluster[0].name} to use keys in compartment ${var.oci_identity.compartment_name} where target.key.id = '${var.oke_kms.key_id}'"
    |----------------
    | oci_identity_dynamic_group.oke-kms-cluster is empty tuple

The given key does not identify an element in this collection value.


Error: Invalid index

  on modules\base\bastion\outputs.tf line 9, in output "bastion_instance_principal_group_name":
   9:   value = oci_identity_dynamic_group.bastion_instance_principal[0].name
    |----------------
    | oci_identity_dynamic_group.bastion_instance_principal is empty tuple

The given key does not identify an element in this collection value.
```

**Comment by hyder** [comment]
Ok, I've been able to reproduce this. Change your node_pool_os_version to "7.7" for the time being. A fix is coming

**Comment by kumar-dhanagopal** [comment]
> 
> 
> Ok, I've been able to reproduce this. Change your node_pool_os_version to "7.7" for the time being. A fix is coming

Thanks! After I made that change, the `data.oci_core_images.latest_images is tuple with 1 element` error doesn't occur.

**Comment by kumar-dhanagopal** [comment]
For the `oci_identity_dynamic_group.oke-kms-cluster is empty tuple` error, I tried setting `statements` in `modules/policies/policies.tf explicitly to `["Allow dynamic-group ${oci_identity_dynamic_group.oke-kms-cluster[0].name} to use keys in compartment ${var.oci_identity.compartment_name} where target.key.id = '${var.oke_kms.key_id}'"]` instead of pointing it to `local.policy_statement`, and the error went away.

**Comment by kumar-dhanagopal** [comment]
And the last error `oci_identity_dynamic_group.bastion_instance_principal is empty tuple` occurs only when `enable_instance_principal = false` in tfvars. It appears that the output `bastion_instance_principal_group_name` in `modules\base\bastion\outputs.tf` needs to be conditionalized to depend on `enable_instance_principal` setting.

**Comment by hyder** [comment]
there's a fix for these in #92. but not tested yet by @narendradotg who reported it.

Can you please try it and see if that fixes it? see in #85 for how to test

**Comment by hyder** [comment]
fixed in #107
